tests/integration/telemetry/stackdriver/stackdriver_filter_audit_test.go

```diff
@@ -0,0 +1,147 @@
+
+			t.Logf("Traffic sent to namespace %v", ns)
+			for _, cltInstance := range clt {
+				scopes.Framework.Infof("Validating Audit policy and Telemetry for Cluster %v", cltInstance)
+				g.Go(func() error {
+					err := retry.UntilSuccess(func() error {
+						if err := sendTrafficForAudit(t); err != nil {
+							return err
+						}
+
+						clName := cltInstance.Config().Cluster.Name()
+						t.Logf("Collect Audit Log for cluster %v", clName)
+
+						errAuditFoo := validateLogs(t, serverAuditFooLogEntry, clName)
+						if errAuditFoo == nil {
+							t.Logf("Foo Audit Log validated for cluster %v", clName)
+						}
+						errAuditBar := validateLogs(t, serverAuditBarLogEntry, clName)
+						if errAuditBar == nil {
+							t.Logf("Bar Audit Log validated for cluster %v", clName)
+						}
+						errAuditAll := validateLogs(t, serverAuditAllLogEntry, clName)
+						if errAuditAll == nil {
+							t.Logf("All Audit Log validated for cluster %v", clName)
+						}
+						errAuditNone := validateLogs(t, serverAuditNoneLogEntry, clName)
+						if errAuditNone == nil {
+							t.Logf("None Audit Log validated for cluster %v", clName)
+						}
+						if errAuditFoo == nil && errAuditBar == nil && errAuditAll == nil && errAuditNone != nil {
+							return nil
+						}
+						return nil
+					}, retry.Delay(5*time.Second), retry.Timeout(80*time.Second))
+
+					if err != nil {
+						return err
+					}
+					return nil
+				})
+			}
+			if err := g.Wait(); err != nil {
+				t.Fatalf("test failed: %v", err)
+			}
+		})
+}
+
+// send http requests with different header and path
+func sendTrafficForAudit(t *testing.T) error {
+	t.Helper()
+
+	newOptions := func(headers http.Header, path string) echo.CallOptions {
+		return echo.CallOptions{
+			Target:   srv[0],
+			PortName: "http",
+			Headers:  headers,
+			Path:     path,
+			Count:    requestCountMultipler * len(srv),
+		}
+	}
+
+	opts := []echo.CallOptions{
+		// request will be logged if "request header" value and "to operation path" is matched with audit policy
+		// path "/audit-none" will be filtered by audit policy and will not be logged
+		newOptions(nil, "/audit-none"),
+		newOptions(map[string][]string{"X-Audit": {"foo"}}, "/audit-none"),
+		newOptions(map[string][]string{"x-Audit": {"bar"}}, "/audit-none"),
+
+		// Headers are case sensitive for this test framework. It requires capitalize the first letter of every word
+		newOptions(map[string][]string{"X-Header": {"bar"}}, "/foo"),
+		newOptions(map[string][]string{"X-Header": {"foo"}}, "/bar"),
+		newOptions(map[string][]string{"X-Header": {"bar"}}, "/bar"),
+		newOptions(map[string][]string{"X-Header": {"foo"}}, "/foo"),
+
+		// path "/audit-all" is matched in audit policy and all requests will be logged
+		newOptions(nil, "/audit-all"),
+		newOptions(map[string][]string{"X-Audit": {"foo"}}, "/audit-all"),
+		newOptions(map[string][]string{"X-Audit": {"bar"}}, "/audit-all"),
+	}
+
+	for _, cltInstance := range clt {
```

Comment:
you can just pass this as param to sendTraffic and not need to run the loop here